### ISSUE TITLE
tensorflow_datasets.image.MNIST overriding to add extra functionality

### ISSUE THREAD

**Issue by SouBanerjee** [tensorflow/datasets#1917]
Hi,
I am new to tensorflow
I was wondering how to override predefined tensorflow_datasets.image.MNIST for my own work.
For example:
    I want to create a split-mnist dataset in which I will be accessing images from only two classes at a single time, how to modify MNIST class or how to inherit MNIST class to add that feature.

In pytorch, I could inherit torchvision.datasets.MNIST into another class and then I could get a mask for the classes for which I need to access the images and then I could get the images of those classes by calling the dataloader and for that I just needed to pass the class_labels for which classes I need images as an argument to the inherited class

**Environment information**
(if applicable)
* Operating System: Ubuntu 18.04
* Python version: 3.7
* `tensorflow-datasets version: 1.2.0
* `tensorflow-gpu version: 2.1.0

**Comment by Conchylicultor** [comment]
Sorry for the late answer. Mnist is small enough so you could load it entirelly in memory and use `ds = ds.filter` to only keep labels satisfying a specific condition.
Alternativelly, you can inherit from MNIST and overwrite `_generate_examples`

```
def _generate_examples(self):
  for key, ex in super()._generate_examples():
    if ex['label'] != 1:
      continue
    yield key, ex
```

Closing this as the issue is old so probably obsolete, but don't hesitate to re-open if you need more info.
